pkg/apis/networking/validation/validation.go

```diff
@@ -721,3 +721,66 @@ func ValidateIPAddressUpdate(update, old *networking.IPAddress) field.ErrorList
 	allErrs = append(allErrs, apivalidation.ValidateImmutableField(update.Spec.ParentRef, old.Spec.ParentRef, field.NewPath("spec").Child("parentRef"))...)
 	return allErrs
 }
+
+var ValidateServiceCIDRName = apimachineryvalidation.NameIsDNSSubdomain
+
+func ValidateServiceCIDR(cidrConfig *networking.ServiceCIDR) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&cidrConfig.ObjectMeta, false, ValidateServiceCIDRName, field.NewPath("metadata"))
+
+	if cidrConfig.Spec.IPv4 == "" && cidrConfig.Spec.IPv6 == "" {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("Spec"), cidrConfig.Spec, "at least one CIDR required"))
```

Comment:
During the next phase we'll have to deal with the problem of the primary IPFamily, we have a notion of primary IP family and the apiservers has to be consistent, since when you create a Service and you don't specify a ClusterIP, it will assign an IP from the primary ... if we do the labels there can be races, the single Name uniqueness iprovide us consistency across all apiservers (that is the trick we are doing with the IPAddresses names) ... we'll have to figure out how to indicate IPv4,IPv6 or IPv6,IPv4 but we can use an annotation for that.

The slice of strings has two drawbacks in my opinion, if we do that we better move to a Singleton object and we allow Updates, but is easier for the user to make mistakes and is harder to implement the orphaning detection logic

Reply:
There can be only one primary IP family and it has to be the same across the cluster and you can not cross validate, so having different objects with different primary IP families is not valid, the same applies to arrays, you need to apply a convention  that is not immediate to know from the API object

From the user POV I think that is easier to understand having fields IPv4 and IPv6 , also if you think in the userspace tools, you already have this distinction ,  `ping` and `ping6` , `iptables` and `ip6tables` ...

Reply:
> you need to apply a convention that is not immediate to know from the API object

Ah, true. But that just means that `primaryCIDR`/`secondaryCIDR` isn't an option here. `[]CIDRs` still is.

> From the user POV I think that is easier to understand having fields IPv4 and IPv6

If you have a dual-stack cluster, then I'd say it's equally easy to understand either way. We already have lots of fields that mix IPv4 and IPv6 IPs together (`podIPs`, `clusterIPs`, `node.status.addresses`) and nobody gets confused by them.

If you _don't_ have a dual-stack cluster, then I think explicit v4/v6 fields are actually harder to understand. Especially, it is harder to document; rather than saying "set `CIDR` to what you want the new CIDR to be", you have to say "if you're in a single-stack IPv4 cluster, then set `IPv4` to what you want the new CIDR to be, and if you're in a single-stack IPv6 cluster, then set `IPv6` to what you want the new CIDR to be."

Reply:
Discussed with Dan offline, for reference, after this PR we still have to solve the problem of defaulting the Primary IPFamily across the cluster, using an Array solves this problem because we can follow the same convention we are using now with the flags.

Since this is alpha, all code is gated and adding the logic to solve the primary ip family in this PR will increase a lot the complexity of the review, we defer this discussion for the next iteration, including the change of the fields by an array

Reply:
@danwinship @khenidak @thockin please comment on the doc https://docs.google.com/document/d/1-CKDPRHsG6XfnGs-Bb7ptWgDtTq9-hj8aJQpGz1CSh8/edit , I think Dan is right and an array limited to only two cidrs max (one IPv4 and other IPv6) is the best alternative for the API. If you can agree today I will update the PR and the KEP this weekend (code freeze is next wednesday)